src/RA_AchievementSet.cpp

```diff
@@ -67,8 +67,7 @@ void AchievementSet::OnRequestUnlocks(const Document& doc)
 {
     if (!doc.HasMember("Success") || doc["Success"].GetBool() == false)
     {
-        ASSERT(!"Invalid unlocks packet!");
-        return;
+        throw std::invalid_argument{"Invalid unlocks packet!"};
```

Comment:
I think the choice to use asserts was intentional. An uncaught throw will terminate the app. I think better error handling/reporting _is_ desirable, but these throws are not.

Reply:
Alright, I'll put it back until we can figure out a way to handle it properly.